lesson3/Lesson3HW/app/src/main/java/com/margaret/lesson3hw/MainActivity.java

```diff
@@ -32,25 +32,35 @@ public void onClick(View view) {
         });
 
         //set up fragment manager stuff
-        final Fragment settingsFragment = new SettingsFragment();
         Fragment defaultFragment = new MainActivityFragment();
-        FragmentManager fragmentManager = getSupportFragmentManager();
-        final FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
+        final FragmentManager fragmentManager = getSupportFragmentManager();
+        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
 
         //set the default to be the list view
-        fragmentTransaction.add(R.id.mainframelayout, defaultFragment);
+        fragmentTransaction.add(R.id.mainframelayout, defaultFragment).commit();
 
         //add onclick to button to switch to settings screen
-        Button settingsButton = (Button) this.findViewById(R.id.settingsbutton);
-        settingsButton.setOnClickListener(new View.OnClickListener() {
+        final Button switchFragmentButton = (Button) this.findViewById(R.id.switchFragmentButton);
+        switchFragmentButton.setOnClickListener(new View.OnClickListener() {
```

Comment:
I made this button a "switch fragment" button instead of just "go to settings" button so that the user can go back to the main fragment.